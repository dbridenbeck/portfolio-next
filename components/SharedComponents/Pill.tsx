import styled from "styled-components";

const PillContainer = styled.div`
  height: 1.75em;
  padding: 0.133em 0.825em;
  margin: ${({ setMargin }) => setMargin ? setMargin : "0 0 0 0" };
  color: #584d4d;
  background-color: #f2f2f2;
  border-radius: 1000px;
  font-size: 0.825em;
`;

interface PillProps {
  children: string;
  setMargin: string;
}

const Pill: React.FC<PillProps> = ({ children, setMargin }) => (
  <PillContainer setMargin={setMargin}>
    {children}
  </PillContainer>
);

export default Pill;
