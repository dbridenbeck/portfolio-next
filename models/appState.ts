export interface ProjectModel {
  title: string;
  type: string;
  tech: string[];
  projectDetail: string;
  url: string;
  color: string;
  webm: string;
  mp4: string;
  open: boolean;
  thumb: string;
  thumbAlt: string;
}

export default interface AppStateModel {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  projects: ProjectModel[];
  pageClickedOnce: boolean;
}
