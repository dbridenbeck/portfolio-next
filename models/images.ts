export default interface ImageModel {
  currentPage: string;
  pageClickedOnce: boolean;
}