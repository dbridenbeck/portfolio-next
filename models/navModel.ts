export interface NavProps {
  currentPage: string;
  pages: [
    { pageName: string; color: string },
    { pageName: string; color: string },
    { pageName: string; color: string }
  ];
  changePage: (text: string) => void;
}
