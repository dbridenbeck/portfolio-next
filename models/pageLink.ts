export default interface PageLink {
  onHelloPage: boolean;
}
