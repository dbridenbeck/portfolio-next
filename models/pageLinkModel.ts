export interface PageLinkProps {
  page: { pageName: string; color: string };
  currentPage: string;
  handleChangePage: (text: string) => void;
}
