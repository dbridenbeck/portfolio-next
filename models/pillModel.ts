export interface PillProps {
  children: string;
  setMargin: string;
}
