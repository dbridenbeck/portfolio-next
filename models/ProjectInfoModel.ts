import { ProjectModel } from './appState';

export interface ProjectInfoProps {
  project: ProjectModel;
}
