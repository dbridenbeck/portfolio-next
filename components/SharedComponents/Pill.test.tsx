import React from "react";
import { render } from "@testing-library/react";
import { PillProps } from "../../models/pillModel";
import Pill from "./Pill";

describe("Pill", () => {
  it("should render without errors", () => {
    const requiredProps: PillProps = {
      children: "React",
      setMargin: "0 0.75em 0 0",
    };
    const { container } = render(<Pill {...requiredProps}/>);
    expect(container).toMatchSnapshot();
  });
});