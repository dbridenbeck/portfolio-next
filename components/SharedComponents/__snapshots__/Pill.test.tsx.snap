// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill should render without errors 1`] = `
<div>
  <div
    class="Pill__PillContainer-c9peiv-0 bvqjVl"
  >
    React
  </div>
</div>
`;
